search-sentiments-map/public/scripts/current-data-script.js

```diff
@@ -65,11 +107,24 @@ function setUserSearchTopic(topic, countries) {
 }
 
 /**
- * Fetches current top trends from the backend and displays them on the website.
+ * Changes `currentTimeRange` parameter and updates trends for new time range.
+ * @param {number} timeRange The interval value for the time range.
  */
-function updateTrends() {
-  fetch('/trends').then(globalTrends => globalTrends.json()).then(trends => {
-    topTrends = trends;
-    setTopTrends(); 
-  });
+function changeTimeRange(timeRange) {
+  currentTimeRange = timeRange;
+  updateTrends();
```

Comment:
do you think you should do a diff against the currentTimeRange and the newly set timeRange? In case there's no difference, then we shouldn't need to update the trends right?